python/cudf/cudf/core/udf/strings_typing.py

```diff
@@ -0,0 +1,242 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+
+import operator
+
+from numba import types
+from numba.core.typing import signature as nb_signature
+from numba.core.typing.templates import AbstractTemplate, AttributeTemplate
+from numba.cuda.cudadecl import registry as cuda_decl_registry
+
+from strings_udf._typing import StringView, string_view
+
+from cudf.core.udf._ops import comparison_ops
+from cudf.core.udf.masked_typing import MaskedType
+
+
+# String functions
+@cuda_decl_registry.register_global(len)
+class MaskedStringViewLength(AbstractTemplate):
+    """
+    provide the length of a cudf::string_view like struct
+    """
+
+    def generic(self, args, kws):
+        if isinstance(args[0], MaskedType) and isinstance(
+            args[0].value_type, StringView
+        ):
+            return nb_signature(MaskedType(types.int32), args[0])
```

Comment:
Long term I'd like to avoid hardcoding `int32` and use an alias for `size_type`.

Reply:
This was done in https://github.com/rapidsai/cudf/pull/11319/commits/9231469a3b188c00e8a8099761e7867e3ffdd1b1.